### PR TITLE
Wallet connect and disconnect template

### PR DESCRIPTION
**Pull request by rmeena840**
Wallet connect and disconnect templat
Screenshot 2022-04-02 at 10 05 21 PM
e

### CODE REVIEW

File: wallet-connect/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        '**/*.test.tsx',
+        '**/*.spec.js',
+        '**/*.spec.jsx',
+        '**/*.spec.tsx',
+      ],
+      env: {
+          jest: true,
+      },
+    },
+  ],
```
Comment (robert-zaremba) [inline wallet-connect/.eslintrc.js]: overrides is not needed, please remove

File: wallet-connect/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    node: true,
+    es6: true,
+    es2020: true,
+    mocha: true
+  },
+  plugins: ["simple-import-sort", "sonarjs"],
+  extends: [
+    "plugin:sonarjs/recommended",
+    "eslint:recommended",
+    "plugin:react/recommended"
```
Comment (robert-zaremba) [inline wallet-connect/.eslintrc.js]: this is not a react project. Where did you copy this file from?

File: wallet-connect/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Wallet-Connect Template
+
+This tutorial will guide you on AlgoSigner wallet connect and disconnect.
```
Comment (robert-zaremba) [inline wallet-connect/README.md]:
```suggestion
This template shows how to connect a AlgoSigner wallet with a Vue.js app.
```

File: wallet-connect/algob.config.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  token: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+  // you can also load accounts from KMD
+  accounts: accounts
+};
+
+module.exports = {
+  networks: {
+    default: defaultCfg
+  }
+};
```
Comment (robert-zaremba) [inline wallet-connect/algob.config.js]: algob.config is not needed. This template doesn't interact with any smart contract.

**Review by robert-zaremba — CHANGES_REQUESTED** [review]
We wanted to use Vue support all wallets.

File: wallet-connect/.editorconfig [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# editorconfig.org
```
Comment (robert-zaremba) [inline wallet-connect/.editorconfig]: please move this file to the root

File: wallet-connect/.eslintignore [hunk 6]
```diff
@@ -0,0 +1,8 @@
+**/node_modules
+**/dist
+**/package-lock.json
+.DS_Store
+**/coverage
+packages/crypto
+**/*.config.js
+.log
```
Comment (robert-zaremba) [inline wallet-connect/.eslintignore]: ditto

File: wallet-connect/.gitignore [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+.vscode
+*.suo
+*.ntvs*
+*.njsproj
+*.sln
+*.sw?
+node_modules
+
+# macOS
+.DS_Store
```
Comment (robert-zaremba) [inline wallet-connect/.gitignore]: we already have most of this stuff in the root .gitignore. please unify

File: wallet-connect/.yarnrc.yml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+nodeLinker: node-modules
+
+pnpMode: loose
+
+yarnPath: .yarn/releases/yarn-3.2.0.cjs
```
Comment (robert-zaremba) [inline wallet-connect/.yarnrc.yml]: this is also already in the root. Why adding another one? We are using monorepos

File: wallet-connect/README.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
+# Wallet-Connect Template
+
+This template shows how to connect a AlgoSigner wallet with a Vue.js app.
+
+# Steps
```
Comment (robert-zaremba) [inline wallet-connect/README.md]:
```suggestion
# Functionality
```

File: wallet-connect/README.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+### `Connect`
+
+Click on Connect button to connect to AlgoSigner. This will dislay wallet address below on successful connection.
+
+### `Disconnect`
+
+Click on Disconnect button to disconnect to AlgoSigner. This will remove wallet address below Connect button and disconnect from wallet.
+
+### `yarn dev`
```
Comment (robert-zaremba) [inline wallet-connect/README.md]:
```suggestion
## Building

### `yarn dev`
```

File: wallet-connect/package.json [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+		"@vue/cli-plugin-eslint": "^5.0.1",
+		"@vue/cli-plugin-typescript": "^5.0.1",
+		"@vue/cli-service": "^5.0.1",
+		"@vue/compiler-sfc": "^3.2.31",
+		"@vue/eslint-config-typescript": "^10.0.0",
+		"eslint": "^8.11.0",
+		"eslint-config-prettier": "^8.5.0",
+		"eslint-plugin-vue": "^8.5.0",
+		"jest": "^27.5.1",
+		"lint-staged": "^12.3.5",
```
Comment (robert-zaremba) [inline wallet-connect/package.json]: this is already in the root

File: wallet-connect/src/assets/logo.svg [hunk 12]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!-- Created with Inkscape (http://www.inkscape.org/) -->
+
+<svg
```
Comment (vuvoth) [inline wallet-connect/src/assets/logo.svg]: Have we use assets/logo.png and logo.svg?

File: wallet-connect/package.json [hunk 13]
```diff
@@ -0,0 +1,56 @@
+{
+	"name": "sigmadao-ui",
+	"description": "Sigma DAO UI",
```
Comment (robert-zaremba) [inline wallet-connect/package.json]: this is wrong, please rename

File: wallet-connect/package.json [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+	"homepage": "https://algobuilder.dev",
+	"license": "mpl-2.0",
```
Comment (robert-zaremba) [inline wallet-connect/package.json]:
```suggestion
	"license": "ISC",
```

File: wallet-connect/package.json [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+	},
+	"scripts": {
+		"dev": "vue-cli-service serve",
+		"build": "vue-cli-service build",
+		"lint": "yarn lint:js && yarn lint:prettier",
+		"start": "node .output/server/index.mjs",
+		"lint:js": "eslint --ext \".js,.ts,.vue\" --ignore-path .gitignore .",
+		"lint:prettier": "prettier --check .",
+		"lint:fix": "prettier --write --list-different . && yarn lint:js --fix",
+		"upgrade:dependency": "npx yarn-upgrade-all",
```
Comment (robert-zaremba) [inline wallet-connect/package.json]: let's remove this. We have monorepo and this should be done by `yarn up` in the root.

File: wallet-connect/package.json [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+	"dependencies": {
+		"@algo-builder/algob": "^3.2.0",
+		"@algo-builder/web": "^3.2.0",
+		"algosdk": "^1.14.0",
+		"core-js": "^3.21.1",
+		"crypto-browserify": "^3.12.0",
+		"pinia": "^2.0.12",
+		"stream-browserify": "^3.0.0",
+		"vue": "^3.2.31",
+		"vue-router": "^4.0.14"
```
Comment (robert-zaremba) [inline wallet-connect/package.json]: is vue-router needed?

**Review by robert-zaremba — DISMISSED** [review]
- [x] please remove assets (logo images) - they are not relevant in this repository
- [x] add more information in README about config and supported wallets
- [x] In the template, we were suppose to support all wallets , not only algosigner. For that, you were suppose to use git version of algo-builder in the package.json (instead of 3.2)

File: wallet-connect/.eslintrc.js (hunk 1 above)
Comment (rmeena840) [reply wallet-connect/.eslintrc.js]: Removed

File: wallet-connect/.eslintrc.js (hunk 2 above)
Comment (rmeena840) [reply wallet-connect/.eslintrc.js]: Converted to vue

File: wallet-connect/README.md (hunk 3 above)
Comment (rmeena840) [reply wallet-connect/README.md]: Done

File: wallet-connect/algob.config.js (hunk 4 above)
Comment (rmeena840) [reply wallet-connect/algob.config.js]: Removed

File: wallet-connect/src/assets/logo.svg (hunk 12 above)
Comment (rmeena840) [reply wallet-connect/src/assets/logo.svg]: Removed

File: wallet-connect/.editorconfig (hunk 5 above)
Comment (rmeena840) [reply wallet-connect/.editorconfig]: Done

File: wallet-connect/.gitignore (hunk 7 above)
Comment (rmeena840) [reply wallet-connect/.gitignore]: Done

File: wallet-connect/.yarnrc.yml (hunk 8 above)
Comment (rmeena840) [reply wallet-connect/.yarnrc.yml]: Removed

File: wallet-connect/package.json (hunk 11 above)
Comment (rmeena840) [reply wallet-connect/package.json]: Removed

File: wallet-connect/package.json (hunk 14 above)
Comment (rmeena840) [reply wallet-connect/package.json]: Done

File: wallet-connect/package.json (hunk 13 above)
Comment (rmeena840) [reply wallet-connect/package.json]: Done

File: wallet-connect/README.md [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+
+Runs the react app in the development mode.\
+Open [http://localhost:8080](http://localhost:8080) to view it in the browser.
+
```
Comment (Megha-Dev-19) [inline wallet-connect/README.md]: Instead of react shouldn't it say vue ..?

File: wallet-connect/package.json (hunk 15 above)
Comment (rmeena840) [reply wallet-connect/package.json]: Done

File: wallet-connect/README.md [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+### `yarn test`
+
+Launches the test runner in the interactive watch mode.\
+See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
+
+### `yarn build`
+
+Builds the app for production to the `build` folder.\
+It correctly bundles React in production mode and optimizes the build for the best performance.
+
```
Comment (Megha-Dev-19) [inline wallet-connect/README.md]: ditto

File: wallet-connect/README.md (hunk 17 above)
Comment (rmeena840) [reply wallet-connect/README.md]: Done

File: wallet-connect/README.md (hunk 18 above)
Comment (rmeena840) [reply wallet-connect/README.md]: Done

File: wallet-connect/package.json (hunk 16 above)
Comment (rmeena840) [reply wallet-connect/package.json]: Removed the dependency on this from the template.

File: wallet-connect/package.json [hunk 19]
```diff
@@ -0,0 +1,46 @@
+{
+	"name": "wallet-connect",
```
Comment (robert-zaremba) [inline wallet-connect/package.json]:
```suggestion
	"name": "wallet-connect template",
```

File: wallet-connect/.eslintignore (hunk 6 above)
Comment (rmeena840) [reply wallet-connect/.eslintignore]: The first 2 points have been resolved. Other wallets have also been added and have also been verified with `yarn link`. The git version method suggested by you didn't work.

File: wallet-connect/README.md [hunk 20]
```diff
@@ -0,0 +1,39 @@
+# Wallet-Connect Template
+
+This template shows how to connect a AlgoSigner wallet with a Vue.js app.
+
+# Config
```
Comment (robert-zaremba) [inline wallet-connect/README.md]:
```suggestion
## Config
```

Comment (robert-zaremba) [reply wallet-connect/README.md]: I was already writing about this: only the title should have level one header.

File: wallet-connect/README.md [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+
+The file present at `/src/constant/index.ts` is used for ledger name (MainNet, TestNet, Private-Net). By default it is being set to MainNet.
+
+# Supported Wallets
```
Comment (robert-zaremba) [inline wallet-connect/README.md]:
```suggestion
## Supported Wallets
```

File: wallet-connect/README.md [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+
+The follwing wallets are supported:
+
+- AlgoSigner
+- My Algo Connect
+- Wallet Connect
+
+# Functionality
```
Comment (robert-zaremba) [inline wallet-connect/README.md]:
```suggestion
## Functionality
```

File: wallet-connect/README.md [hunk 23]
```diff
@@ -0,0 +1,39 @@
+# Wallet-Connect Template
```
Comment (robert-zaremba) [inline wallet-connect/README.md]:
```suggestion
# Wallet Template
```

**Review by robert-zaremba — COMMENTED** [review]
Please rename wallet-connect directory to `wallet`.

**Review by Megha-Dev-19 — APPROVED** [review]
Lgtm

**Pull request merged**